content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-node-definition-reference.md

```diff
@@ -0,0 +1,382 @@
+	<Branch Name="Success">
+        <Return Name="Entity Spawn Ticket" Type="AzFramework::EntitySpawnTicket"/>
+    </Branch>
+    <Branch Name="Failed"/>
+</Input>
+```
+The above node's C++ code would be used in the following way:
+```cpp
+void CreateSpawnTicketNodeable::CreateTicket(const AzFramework::Scripts::SpawnableScriptAssetRef& prefab)
+{
+    auto ticket = m_spawnableScriptMediator.CreateSpawnTicket(prefab);
+    if (ticket.IsSuccess())
+    {
+        CallSuccess(ticket.GetValue());
+    }
+    else
+    {
+        AZLOG_ERROR("Unable to Create Spawn Ticket - A valid prefab was not provided");
+        CallFailed();
+    }
+}
+```
+----
+
+## Functions
+
+Functions are a way to expose C++ functions directly into the Behavior Context. This results in a way quick and easy way to create libraries of functions that are available for use in Script Canvas.
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Name__ | Required | The name of the function | `Name="RandomColor"` |
+| __Branch__ | Optional | Specifies that the function be used to evaluate the branching condition | `Branch="IsValidFindPosition"` |
+| __BranchWithValue__ | Optional |  | `BranchWithValue="True"` |
+
+Functions can use the [Parameter](#propertydata) element to specify data inputs.
+
+```xml
+<Function Name="RandomColor">
+    <Parameter Name="MinValue" DefaultValue="Data::ColorType(0.0f, 0.0f, 0.0f, 1.0f)"/>
+    <Parameter Name="MaxValue" DefaultValue="Data::ColorType(1.0f, 1.0f, 1.0f, 1.0f)"/>
+</Function>
+
+<Function Name="IsValidFindPosition"> <!-- Used by ContainsString to evaluate the condition to branch -->
+    <Parameter Name="Position" Description="The string find position to check against"/>
+</Function>
+
+<Function Name="ContainsString" Branch="IsValidFindPosition" BranchWithValue="True">
+    <Parameter Name="Source" Description="The string to search in."/>
+    <Parameter Name="Pattern" Description="The substring to search for."/>
+    <Parameter Name="Search From End" Description="Start the match checking from the end of a string."/>
+    <Parameter Name="Case Sensitive" Description="Take into account the case of the string when searching."/>
+</Function>
+
+```
+
+Functions can use the __Out__ element to specify the name for each of the [Branch](#branch) results.
+
+__NOTE:__ There may only be 2 __Out__ elements for branching functions, if more than 2 exist an error will be displayed during compilation.
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Name__ | Required | The name of the function | `<Function Name="Clamp">` |
+
+```xml
+<Function Name="BranchExample" Branch="IsPositive" BranchWithValue="True">
+    <Out Name="Is Positive"/>
+    <Out Name="Is Negative"/>
+</Function>
+```
+----
+
+## Library
```

Comment:
If `Library` is used exclusively with free function nodes, include them under "Free function elements".
```suggestion

### Library
```